### ISSUE TITLE
Dry::Struct::Constructor does not respond to `#meta`

### ISSUE THREAD

**Issue by kj** [dry-rb/dry-struct#130]
**Describe the bug**

When using a custom constructor, the resulting `Dry::Struct::Constructor` doesn't implement `#meta`, so it doesn't work as a type, such as when passed to an `#attribute` of a struct.

**To Reproduce**

```
class Foo < Dry::Struct; end
Foo.meta
# -> {}
FooType = Foo.constructor { Foo.new } # or Types.Constructor(Foo) { Foo.new }
FooType.meta
# NoMethodError (undefined method `meta' for #<Dry::Struct::Constructor type=Foo>)
```

**Expected behavior**

The custom constructor responds to `#meta` like the original struct, and like a custom type should (when using Types.Constructor).

**Your environment**

- Affects my production application: **YES**
- Ruby version: 2.6.2p47
- dry-struct version: 1.1.1
- dry-types version: 1.2.0
- OS: Arch Linux

**Comment by flash-gordon** [comment]
It came as a surprise to me people pass struct classes to `Types::Constructor`, there was no need for that at all. Now dry-types 1.2.2 is out that returns the struct class back if no constructor block is given. This should fix the problems you had. That said, `Struct::Constructor` is still missing some methods from the type interface, we'll eventually add them, I think.

**Comment by kj** [comment]
Hi @flash-gordon, sorry I guess my example was too stripped back. I'm not actually passing it like that (it was just meant to be the quickest way to reproduce the issue), I'm using the constructor block to change the way the arguments are passed to the struct, so I don't think this fixes it for me.

**Comment by flash-gordon** [comment]
fixed in #147
